### DISCUSSION TITLE
How can one formulate a polygon ( particulally pentagon) as obstacle in Casadi. (opti)

### DISCUSSION THREAD

**Discussion by kobbie-hub** [casadi/casadi#3255] [Q&A]
Hello,

How can you formulate a polygon (particularly a pentagon) as an obstacle in Casadi? The aim is to move create a trajectory avoiding the polygon obstacle.
I tried to use Opti to provide inequalities (equation of the line (edges), but returns as an infeasible solution.  I tried to compare at each iteration the points X, Y positions if they lie in polygon regions using the Matlab function and apply this to opti.subject_to as a logic condition, but not successful.  Any advice, guidance, or examples on how best this can be done to include polygon obstacles into the opti framework would be appreciated. Thank you.

**Comment by jgillis** [comment]
Building logic conditions is a non-starter since these lead to zero gradients. A strategy I've seen used a lot is a separating hyperplane. Parametrize the location of the hyperplane as decision variables (e.g. normal vector and offset) and add two constraints: a constraint that all vertices of the polygon should be on one side of the plane, and a constraint that all vertices of the self are on the other.
Other strategies could be using a (signed) distance function

**Reply by kobbie-hub** [reply]
Thanks Joris. I will try the suggestions. ( Still learning casadi)

@jgillis: In one of your replies on the casadi user group page, you made mention of a workshop about this by your research group regarding encoding obstacles. I want to find out if this specific workshop on obstacles is available or recorded for training that I can access.

-------------------------------------------------------------------------------------------------------------------------------------------------------
https://groups.google.com/g/casadi-users/c/rEkSwnq9uG8

**Reply by TarasLashiv** [reply]
Доброго дня! Розбираюсь у вашому питанні
